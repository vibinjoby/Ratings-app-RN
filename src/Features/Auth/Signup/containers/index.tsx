import React from 'react'
import { useNavigation } from '@react-navigation/native'
import { useFormik } from 'formik'

import { ScreenNames, SignUpValidationSchema } from '../../constants'
import { ScreenNames as BaseModuleScreenNames } from '../../../../BaseModule/constants'
import { ScreenNames as HomeScreenNames } from '../../../Home/constants'
import { RegisterProps } from '../../types'
import { useRegister } from '../../hooks'
import SignUpForm from '../components/SignUpForm'
import { determineUserTypeByTabSelection } from '../../helpers'
import ApiResult from '../../../../components/ApiResult'
import { decodeAndSaveToken } from '../../../../utilities/helpers'
import { UserType } from '../../../../../__generated__/globalTypes'

const SignUp: React.FC = () => {
  const navigation = useNavigation()

  const { values, errors, setFieldValue, isValid, dirty } = useFormik<RegisterProps>({
    initialValues: {
      fullName: '',
      email: '',
      password: '',
      confirmPassword: '',
      selectedTab: 0,
    },
    validationSchema: SignUpValidationSchema,
    onSubmit: () => {},
  })

  const { onRegisterMutate, loading, error } = useRegister({
    email: values['email'],
    password: values['password'],
    fullName: values['fullName'],
    userType: determineUserTypeByTabSelection(values['selectedTab']),
  })

  const handleSignUp = async () => {
    const { data } = await onRegisterMutate()
    const userType = determineUserTypeByTabSelection(values['selectedTab'])
    decodeAndSaveToken(data?.createUser?.token, false)

    // Navigate to home based on user type
    navigation.reset({
      index: 0,
      routes: [
        {
          name: BaseModuleScreenNames.HOME_STACK,
          params: {
            initialRoute:
              userType === UserType.customer
                ? HomeScreenNames.CUSTOMER_HOME
                : userType === UserType.owner
                ? HomeScreenNames.OWNER_HOME
                : HomeScreenNames.ADMIN_HOME,
          },
        },
      ],
    })
  }

  const handleSignIn = () => navigation.navigate(ScreenNames.LOGIN)

  return (
    <ApiResult loading={loading} error={error}>
      <SignUpForm
        onSignIn={handleSignIn}
        onSignUp={handleSignUp}
        values={values}
        errors={errors}
        isValid={isValid}
        dirty={dirty}
        handleChange={(fieldName: string) => (e: React.ChangeEvent) => {
          setFieldValue(fieldName, e)
        }}
      />
    </ApiResult>
  )
}

export default SignUp
