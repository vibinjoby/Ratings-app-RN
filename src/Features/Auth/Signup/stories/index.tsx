export { default } from './SignUp.story'
