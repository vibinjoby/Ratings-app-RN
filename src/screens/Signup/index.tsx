import { useNavigation } from '@react-navigation/native'
import React, { useState } from 'react'
import { View, Text, ImageBackground, Image, TouchableOpacity, ScrollView } from 'react-native'

import Button from '../../components/Button'
import TextField from '../../components/TextField'
import TwoTabs from '../../components/TwoTabs'
import routes from '../../navigations/routes'
import Typography from '../../utilities/typography'
import styles from './styles'

const SignUp: React.FC = () => {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [cnfrmPassword, setCnfrmPassword] = useState('')
  const [selectedTab, setSelectedTab] = useState(0)
  const navigation = useNavigation()

  return (
    <ScrollView>
      <View style={styles.container}>
        <ImageBackground
          style={styles.bgContainer}
          source={require('../../assets/authBg/authBg.png')}
        >
          <Image source={require('../../assets/loginDish/loginDish.png')} />
        </ImageBackground>
        <View style={styles.contentContainer}>
          <Text style={Typography.Title1.regular}>Sign Up</Text>
          <View style={styles.tabContainer}>
            <TwoTabs
              selectedTab={selectedTab}
              tab1Text="Customer"
              tab2Text="Owner"
              onTabSelect={(tab) => setSelectedTab(tab)}
            />
          </View>

          <TextField
            testID="email"
            containerStyle={styles.emailField}
            textHint="Email/Username"
            inputValue={email}
            onInputChange={(e) => setEmail(e)}
          />
          <TextField
            testID="password"
            containerStyle={styles.pwdField}
            textHint="Password"
            isProtected
            inputValue={password}
            onInputChange={(e) => setPassword(e)}
          />
          <TextField
            testID="confirmPassword"
            containerStyle={styles.pwdField}
            textHint="Confirm Password"
            isProtected
            inputValue={cnfrmPassword}
            onInputChange={(e) => setCnfrmPassword(e)}
          />

          <Button
            testID="signupBtn"
            disabled={!email || !password}
            title="Sign Up"
            onPress={() => console.log('pressed')}
            customStyle={styles.loginBtn}
          />
          <View style={styles.footerContainer}>
            <Text style={styles.footerTxt}>Already have an account? </Text>

            <TouchableOpacity
              activeOpacity={0.7}
              // @ts-ignore
              onPress={() => navigation.navigate(routes.LOGIN)}
            >
              <Text style={styles.signupTxt}>Sign In</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    </ScrollView>
  )
}

export default SignUp
