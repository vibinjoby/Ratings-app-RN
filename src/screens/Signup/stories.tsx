import React from 'react'
import { storiesOf } from '@storybook/react-native'

import SignUp from '.'
import { Provider } from 'react-redux'
import { store } from '../../store'
import { NavigationContainer } from '@react-navigation/native'

storiesOf('Screens', module).add('Signup', () => (
  <NavigationContainer>
    <Provider store={store}>
      <SignUp />
    </Provider>
  </NavigationContainer>
))
