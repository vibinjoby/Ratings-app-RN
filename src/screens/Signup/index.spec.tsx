import 'react-native'
import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import { NavigationContainer } from '@react-navigation/native'

import Signup from '.'

describe('Sign Up screen', () => {
  it('should render', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <Signup />
      </NavigationContainer>,
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('should let me enter values and submit', () => {
    const { getByTestId } = render(
      <NavigationContainer>
        <Signup />
      </NavigationContainer>,
    )
    fireEvent.changeText(getByTestId('email'), 'Email')
    fireEvent.changeText(getByTestId('password'), 'Password')
    fireEvent.changeText(getByTestId('confirmPassword'), 'Confirm Password')
    fireEvent.press(getByTestId('signupBtn'))
  })
})
