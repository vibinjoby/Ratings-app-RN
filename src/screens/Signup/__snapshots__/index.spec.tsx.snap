// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up screen should render 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "alignItems": "center",
            "height": 240,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/authBg/authBg.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 240,
                "width": "100%",
              },
              undefined,
            ]
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/loginDish/loginDish.png",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#181725",
              "fontFamily": "Poppins-Regular",
              "fontSize": 26,
              "lineHeight": 29,
            }
          }
        >
          Sign Up
        </Text>
        <View
          style={
            Object {
              "alignSelf": "center",
              "marginTop": 15,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomLeftRadius": 40,
                      "borderTopLeftRadius": 40,
                      "marginRight": 2,
                    },
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#E2E2E2",
                      "borderColor": "#fff",
                      "borderWidth": 1,
                      "elevation": 3,
                      "paddingHorizontal": 14,
                      "paddingVertical": 8,
                      "shadowColor": "#181725",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowOpacity": 0.22,
                      "shadowRadius": 2.22,
                    },
                    Object {
                      "backgroundColor": "#FF5823",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins-Regular",
                        "fontSize": 16,
                      },
                      Object {
                        "color": "#fff",
                      },
                    ]
                  }
                >
                  Customer
                </Text>
              </View>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomRightRadius": 40,
                      "borderTopRightRadius": 40,
                    },
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#E2E2E2",
                      "borderColor": "#fff",
                      "borderWidth": 1,
                      "elevation": 3,
                      "paddingHorizontal": 14,
                      "paddingVertical": 8,
                      "shadowColor": "#181725",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowOpacity": 0.22,
                      "shadowRadius": 2.22,
                    },
                    false,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins-Regular",
                        "fontSize": 16,
                      },
                      false,
                    ]
                  }
                >
                  Owner
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 30,
              "marginTop": 40,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#7C7C7C",
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
              }
            }
          >
            Email/Username
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#E2E2E2",
                  "color": "#181725",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 18,
                  "paddingVertical": 10,
                  "width": "100%",
                }
              }
              testID="email"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#7C7C7C",
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
              }
            }
          >
            Password
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#E2E2E2",
                  "color": "#181725",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 18,
                  "paddingVertical": 10,
                  "width": "100%",
                }
              }
              testID="password"
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                }
              }
              testID="eyeIc"
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/assets/password/password.png",
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#7C7C7C",
                "fontFamily": "Poppins-Regular",
                "fontSize": 16,
              }
            }
          >
            Confirm Password
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              autoCapitalize="none"
              onChangeText={[Function]}
              secureTextEntry={true}
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#E2E2E2",
                  "color": "#181725",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 18,
                  "paddingVertical": 10,
                  "width": "100%",
                }
              }
              testID="confirmPassword"
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                }
              }
              testID="eyeIc"
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../src/assets/password/password.png",
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#E2E2E2",
              "borderRadius": 19,
              "marginTop": 40,
              "opacity": 1,
              "padding": 24,
            }
          }
          testID="signupBtn"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 18,
                },
                Object {},
              ]
            }
          >
            Sign Up
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 25,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Poppins-Regular",
                "fontSize": 14,
              }
            }
          >
            Already have an account? 
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#54BF29",
                  "fontFamily": "Poppins-Regular",
                  "fontSize": 14,
                }
              }
            >
              Sign In
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
